### PR TITLE
Attribute waste emissions to energy-related emissions of sectors based on historical shares and remove waste emissions variables from energy-related emissions taxonomy

### PR DESCRIPTION
**Pull request by fschreyer**
The change addresses [this issue](https://github.com/remindmodel/development_issues/issues/261) by attributing the newly introduced waste emissions in REMIND to sectors via a post-processing in the reportEmi() function. Waste emissions are thereby removed from taxonomy of energy emissions (variables with ``+``), such that e.g. ``Emi|CO2|Energy|+|Waste (Mt CO2/yr)`` becomes ``Emi|CO2|Energy|Waste (Mt CO2/yr)``. 
 
The attribution to sectors happens in the following way:

* ``Emi|CO2|Energy|Waste|Plastics Incineration`` is distributed across ``Supply|Electricity``, ``Supply|Heat``, ``Demand|Industry`` and ``Demand|Buildings`` based on 2019 historical shares of waste energy use of the sectors from IEA data (derived by [this function](https://github.com/pik-piam/mrremind/blob/master/R/calcWasteEnergyUseShares.R) from @fbenke-pik ). 

* ``Emi|CO2|Energy|Waste|Feedstocks unknown fate (Mt CO2/yr)`` is attributed to industry in line with the accounting we had before the explicit feedstocks implementation. It refers to emissions from non-plastic materials. Note that accounting of these emissions is currently switched off in REMIND by default (cm_feedstockEmiUnknownFate is off) such that this only applies if chosen intentionally. In case of industry, waste emissions were attributed to subsectors (steel, cement etc.) by the FE solids share of a subsector in total industry solids. 

These are the reported waste emissions variables with this change:

```
> grep("Emi\\|CO2\\|.*Waste", getNames(testReport, dim=3), value=T)
 [1] "Emi|CO2|pre-CCS|Energy|Demand|Industry|Other Industry|Solids|Waste (Mt CO2/yr)"
 [2] "Emi|CO2|Energy|Waste|Plastics Incineration (Mt CO2/yr)"                        
 [3] "Emi|CO2|Energy|Waste|Feedstocks unknown fate (Mt CO2/yr)"                      
 [4] "Emi|CO2|Energy|Waste (Mt CO2/yr)"                                              
 [5] "Emi|CO2|Energy|Supply|Waste (Mt CO2/yr)"                                       
 [6] "Emi|CO2|Energy|Demand|Waste (Mt CO2/yr)"                                       
 [7] "Emi|CO2|Energy|Supply|Electricity|Waste (Mt CO2/yr)"                           
 [8] "Emi|CO2|Energy|Supply|Heat|Waste (Mt CO2/yr)"                                  
 [9] "Emi|CO2|Energy|Demand|Industry|Waste (Mt CO2/yr)"                              
[10] "Emi|CO2|Energy|Demand|Buildings|Waste (Mt CO2/yr)"                             
[11] "Emi|CO2|Energy|Demand|Industry|Solids|+|Waste (Mt CO2/yr)"                     
[12] "Emi|CO2|Energy|Demand|Industry|Other Industry|Solids|+|Waste (Mt CO2/yr)"      
[13] "Emi|CO2|Cumulated|Energy|Waste (Mt CO2)" 
```

It was checked that all variable summations affected by this change are fulfilled. Moreover, waste emissions were added to the cs2 plots. 

See example scenario plots: ``/p/projects/ariadne/remind/compScen-Check_WasteEmi_Reporting-2024-08-12_13.56.28-H12.pdf``. 

Tagging @robinhasse and @orichters fyi.

### CODE REVIEW

**Comment by orichters** [comment]
A challenge that I see is that the [fix for NAVIGATE](https://github.com/pik-piam/piamInterfaces/blob/fa7f63eb47275512c431011ea7a1fddfed4915fd/inst/mappings/mapping_NAVIGATE.csv#L2-L9) and similar for AR6 will then yield the wrong result. But I would also prefer not to take it out because ideally piamInterfaces yields correct results also with older mif files. 

The a bit ugly solution I see is subtracting `Emi|CO2|Energy|Waste|Feedstocks unknown fate` and `Emi|CO2|Energy|Waste|Plastics Incineration` again from the variables, neutralizing the fix for people that have the new reporting.

Then, after the next release is out, of course, kick these fixes all out.
Best,
Oliver

**Comment by orichters** [comment]
I agree with Michaja on this one. `Emi|CO2|Energy Supply and Demand|Waste`? That would be very transparent and also avoid the piamInterfaces mess.

We have such variables sometimes, such `Emi|OC|Energy Supply and Demand`.

Which leads me to the question why it is `Emi|CO|Energy Demand|Buildings` but `Emi|CO2|Energy|Demand|Buildings`, notice the difference between using " " or "|" between Energy and Demand. So maybe rather use `Emi|CO2|Energy|Supply and Demand|Waste` and adapt that for the rest as well?

**Comment by fschreyer** [comment]
> Is my understanding correct that the Emi|CO2|Energy|Waste equals the sum of Emi|CO2|Energy|Supply|Waste and Emi|CO2|Energy|Demand|Waste? 
Yes. 

> The IEA has "MEMO" categories for that, maybe we should do that too.

Sounds like a larger project. 

> I agree with Michaja on this one. Emi|CO2|Energy Supply and Demand|Waste? That would be very transparent and also avoid the piamInterfaces mess.

Not sure I understand this right. So, you would like to rename ``Emi|CO2|Energy|Waste`` to ``Emi|CO2|Energy Supply and Demand|Waste`` because people with then understand more easily that  ``Emi|CO2|Energy|Waste`` is contained in ``Emi|CO2|Energy|Waste|Supply`` and  ``Emi|CO2|Energy|Waste|Demand`` and not a separate item. Well, then we would also need to come up with something for ``Emi|CO2|Energy|Supply|Electricity+Heat...`` to make clear that ``Supply|Waste`` is included there. That does not seem practical. I guess we'd rather need Memo items as Michaja suggested. 


Generally, people "consuming" REMIND mif-files without pluses (or without knowing what they mean) can, in theory, anyways only guess taxonomies. Like from the naming structure they would not know that they should not add up ``Energy|Demand|Solids`` and ``Energy|Demand|Industry``. I am unsure of whether our naming decision here be of any help to their hard lifes. I think staying close to the IAMC naming standard (``Energy|Waste``, ``Energy|Demand|Waste`` ...) is the most useful as this is what people in our group are used to. 


> Which leads me to the question why it is Emi|CO|Energy Demand|Buildings but Emi|CO2|Energy|Demand|Buildings, notice the difference between using " " or "|" between Energy and Demand.

The power of choosing variable names when there is no standard.

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]
> For checking the sectoral contributions, sectoral waste emissions variables were added. In case of industry, waste plastic emissions were accounted to the "Other Industry" sector. This is a simplification but given the small contributions of waste emissions overall, a further subdivision of waste emission into industry sectors does not appear to be necessary.

I think this is severely skewing the emissions and emission intensities of the industry subsectors, especially early on, where people in e.g. the ARIADNE project will look closest.

EU27_NZ_CCS100Mt 2025  | DEU (Mt) | DEU (%) | World (Mt) | World (%) |
-----------------------|----------|---------|------------|-----------|
Cement                 |     12.0 |   9.5 % |        821 |    9.8 %  |
Chemicals              |     32.5 |  26 %   |       1634 |   19 %    |
Steel                  |     30.5 |  24 %   |       2201 |   26 %    |
Other Industry         |     32.8 |  26 %   |       2965 |   35 %    |
Waste                  |     18.7 |  15 %   |        795 |    9.4 %  |
Other Industry + Waste |     51.5 |  41 %   |       3760 |   45 %    |

All waste is deemed to be solids (OK), but Other Industry is the subsector with the second lowest share of solids use.  The most likely candidate would be Cement in my view, which is known to gobble up everything that is cheap and burns.  Even steel throws some old tyres into blast furnaces.  So this is something that might come back to bite us.
image

What is troublesome analytically is that there is no energy flow associated with Emi|CO2|**Energy|Demand**|Industry|Other Industry|Solids|+|Waste.  I know it is beyond the scope of this fix, but this is skewing the carbon intensity of the Other Industry sector, which will require extra work to sort out if somebody is looking at that.

**Review by orichters — APPROVED** [review]
Sounds plausible. So we have to live with mapping fixes for a while, but we will survive that.

File: R/reportEmi.R [hunk 1]
```diff
@@ -353,6 +353,154 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
...
+        as.magpie()
+
+    # if H12 regionmapping
+    } else {
+
+      WasteShares <- read.csv(system.file("extdata", "WasteShares_H12.cs4r", package = "remind2"),
+                              sep = ",", skip = 4, header = F) %>%
+        as.magpie()
+
+      }
```
[the remark this answers is not in the document]
Comment (fschreyer) [reply R/reportEmi.R]: Included.

**Comment by fschreyer** [comment]
Attribution of waste emissions to industry subsectors is now based on sectoral solids share in total industry solids to minimize the issues raised above.

**Pull request merged**